db-commons/src/main/scala/org/bitcoins/db/AppLoggers.scala

```diff
@@ -0,0 +1,214 @@
+package org.bitcoins.db
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.classic.LoggerContext
+import ch.qos.logback.classic.encoder.PatternLayoutEncoder
+import ch.qos.logback.core.rolling.RollingFileAppender
+import ch.qos.logback.core.rolling.TimeBasedRollingPolicy
+import ch.qos.logback.core.encoder.Encoder
+import ch.qos.logback.core.FileAppender
+import ch.qos.logback.core.ConsoleAppender
+import ch.qos.logback.classic.joran.JoranConfigurator
+
+/** Provides logging functionality for Bitcoin-S
+  * app modules (i.e. the modules that are capable
+  * of running on their own) */
+private[bitcoins] object AppLoggers {
+
+  sealed private trait LoggerKind
+  private object LoggerKind {
+    case object P2P extends LoggerKind
+    case object ChainVerification extends LoggerKind
+    case object KeyHandling extends LoggerKind
+    case object Wallet extends LoggerKind
+    case object Http extends LoggerKind
+    case object Database extends LoggerKind
+  }
+
+  /**
+    * @return the peer-to-peer submobule logger
+    */
+  def getP2PLogger(implicit conf: AppConfig) = getLoggerImpl(LoggerKind.P2P)
+
+  /**
+    * @return the chain verification submobule logger
+    */
+  def getVerificationLogger(implicit conf: AppConfig): Logger =
+    getLoggerImpl(LoggerKind.ChainVerification)
+
+  /**
+    * @return the key handling submobule logger
+    */
+  def getKeyHandlingLogger(implicit conf: AppConfig): Logger =
+    getLoggerImpl(LoggerKind.KeyHandling)
+
+  /**
+    * @return the generic wallet logger (i.e. everything not related to key handling)
+    */
+  def getWalletLogger(implicit conf: AppConfig): Logger =
+    getLoggerImpl(LoggerKind.Wallet)
+
+  /**
+    * @return the HTTP RPC server submobule logger
+    */
+  def getHttpLogger(implicit conf: AppConfig): Logger =
+    getLoggerImpl(LoggerKind.Http)
+
+  /**
+    * @return the database interaction logger
+    */
+  def getDatabaseLogger(implicit conf: AppConfig): Logger =
+    getLoggerImpl(LoggerKind.Database)
+
+  private val context = {
+    val context = LoggerFactory.getILoggerFactory() match {
+      case ctx: LoggerContext => ctx
+      case other              => sys.error(s"Expected LoggerContext, got: $other")
+    }
+
+    // following three lines prevents Logback from reading XML conf files
```

Comment:
So does this mean our dev flow has to go through these loggers to? Or is this only applicable when the app is running?

Reply:
Both tests and running the app will log to the same place. By default that's both the console and the log file, so this doesn't really change development flow